src/components/tree-item/tree-item.tsx

```diff
@@ -266,11 +277,27 @@ export class TreeItem
           [CSS_UTILITY.rtl]: rtl
         }}
         icon={selectedIcon}
-        scale="s"
+        scale={this.scale === "l" ? "m" : "s"}
       />
     ) : null;
 
     const hidden = !(this.parentExpanded || this.depth === 1);
+    const { hasEndActions } = this;
+    const slotNode = (
+      <slot
+        key="actionsEndSlot"
+        name={SLOTS.actionsEnd}
+        onSlotchange={this.actionsEndSlotChangeHandler}
+      />
+    );
+    const iconStartEl = (
+      <calcite-icon
+        class={{ [CSS.tabTitle]: true, [CSS.iconStart]: true }}
+        flipRtl={this.iconFlipRtl === "start" || this.iconFlipRtl === "both"}
+        icon={this.iconStart}
+        scale={this.scale === "l" ? "m" : "s"}
```

Comment:
Sidebar: I think we can extract some utils to have consistent icon scale applied within components.

Reply:
I feel like extracting utils is becoming my favorite thing to do. I'll open an issue.